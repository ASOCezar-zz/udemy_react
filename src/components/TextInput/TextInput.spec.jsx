import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { TextInput } from '.';

describe('<TextInput />', () => {
  it('should have a value of searchValue', () => {
    const fn = jest.fn();

    render(<TextInput handleChange={fn} searchValue={'teste'} />);

    const input = screen.getByPlaceholderText(/type your search/i);

    expect(input).toBeInTheDocument();

    expect(input.value).toBe('teste');
  });

  it('should call handleChange for each key pressed', () => {
    const fn = jest.fn();

    render(<TextInput handleChange={fn} searchValue="valor qualquer" />);

    const input = screen.getByPlaceholderText(/type your search/i);

    const value = 'o valor';

    userEvent.type(input, value);

    expect(input.value).toBe('valor qualquer');
    expect(fn).toHaveBeenCalledTimes(value.length);
  });

  it('should match with snapshot', () => {
    const fn = jest.fn();
    const { container } = render(<TextInput handleChange={fn} searchValue={'teste'} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
