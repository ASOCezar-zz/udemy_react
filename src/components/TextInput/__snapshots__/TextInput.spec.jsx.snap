// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should match with snapshot 1`] = `
<input
  class="sc-bdnxRM gLqrLJ text-input"
  placeholder="Type your search"
  type="search"
  value="teste"
/>
`;
